fix(RadioButtonGroup): make selected prop reactive

weaseldotro opened the pull request:
Fixes #406

metonym reviewed (changes requested):
Nice work.

One requested change. Programmatically setting and binding `selected` works. However, if the user clicks one of the radio buttons, the update method from the "RadioButtonGroup" context must update `selected` and not `selectedValue`.

Otherwise, there may be two or more selected buttons:

<img width="500" alt="Screen Shot 2020-11-22 at 6 23 04 AM" src="https://user-images.githubusercontent.com/10718366/99906353-7df1b180-2c8b-11eb-9e14-a26262d51042.png">

```diff
setContext("RadioButtonGroup", {
    selectedValue,
    add: ({ checked, value }) => {
      if (checked) {
        selectedValue.set(value);
      }
    },
    update: (value) => {
+      selected = value;
-       selectedValue.set(value);
    },
  });
```

weaseldotro commented:
Oops, somehow I missed testing that. I have added the requested changes.

To correctly test this, in my test program https://codesandbox.io/s/loving-grass-p466p?file=/App.svelte I have replaced ```<RadioButtonGroup selected={plan}>``` with ```<RadioButtonGroup bind:selected={plan}>``` to use two way binding, otherwise it won't work correctly when using both radios and buttons to change the options.
